### ISSUE TITLE
Apple silicon support

### ISSUE THREAD

**Issue by MuMrj** [magicien/JoyKeyMapper#12]
It seems like the app doesn't work with apple silicon devices, I hope support is coming soon.

**Comment by NullYing** [comment]
use Xcode rebuild this code, it's available with apple silicon

**Comment by SrBrits** [comment]
> use Xcode rebuild this code, it's available with apple silicon

Can you explain how to do that? Thanks in advance! Im super new to using GitHub and Xcode.

**Comment by metawops** [comment]
Here on my M1 Max MBP the app basically works. But holds the connection to the controller only a few minutes. After that nothing works anymore and the app has to be restarted – which makes it practically unusable, sadly. 😞

**Comment by volaix** [comment]
> use Xcode rebuild this code, it's available with apple silicon

I tried but there's a bunch of errors?
<img width="885" alt="image" src="https://user-images.githubusercontent.com/16506248/190222664-76a504b8-c425-45c6-b609-382b45466602.png">

**Comment by blackshine2018** [comment]
> Here on my M1 Max MBP the app basically works. But holds the connection to the controller only a few minutes. After that nothing works anymore and the app has to be restarted – which makes it practically unusable, sadly. 😞

do you find any alternative or solution? I am in the same situation.

**Comment by metawops** [comment]
> > Here on my M1 Max MBP the app basically works. But holds the connection to the controller only a few minutes. After that nothing works anymore and the app has to be restarted – which makes it practically unusable, sadly. 😞
> 
> do you find any alternative or solution? I am in the same situation.

Nope, unfortunately not. Nobody else seems to have made something like this and the original author @magicien doesn't seem to support it anymore. 😞

**Comment by metawops** [comment]
Found the solution! 😊 
Here's a working version for Apple Silicon Macs. 👉 https://github.com/qibinc/JoyMapperSilicon
